src/main/java/ladder/domain/Users.java

```diff
@@ -0,0 +1,47 @@
+package ladder.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Users {
+
+    private static final int MINIMUM_SIZE = 2;
+    private static final String USERS_SIZE_ERROR_MESSAGE = "유저는 한명보다 많아야 합니다.";
+    private static final String USERS_DUPLICATED_ERROR_MESSAGE = "중복된 이름을 입력할 수 없습니다.";
+    private final List<User> users;
+
+    public Users(List<String> userNames) {
+        validate(userNames);
+        List<User> collect = userNames.stream()
+                .map(User::new)
+                .collect(Collectors.toList());
+        this.users = new ArrayList<>(collect);
+    }
+
+    private void validate(List<String> userNames) {
+        validateDuplication(userNames);
+
+        validateSize(userNames);
+    }
+
+    private void validateSize(List<String> userNames) {
+        if (userNames.size() < MINIMUM_SIZE) {
+            throw new IllegalArgumentException(USERS_SIZE_ERROR_MESSAGE);
+        }
+    }
+
+    private void validateDuplication(List<String> userNames) {
+        int inputNamesSize = userNames.size();
+        long unDuplicationNamesSize = userNames.stream()
+                .distinct()
+                .count();
+        if (inputNamesSize != unDuplicationNamesSize) {
+            throw new IllegalArgumentException(USERS_DUPLICATED_ERROR_MESSAGE);
+        }
+    }
+
+    public List<User> getUsers() {
+        return users;
```

Comment:
방어적 복사 개념에 대해 알고 계신가요?
```suggestion
        return Collections.unmodifiableList(users);
```
방어적 복사에 대해 학습해보시고 ArrayList로 선언된 기타 객체들에 대해서 "방어적 복사"로 반환해주세요!
(그대로 반환하면 어떤 문제가 있을까요?)

Reply:
컬렉션 반환 메서드들 방어적 복사로 수정했습니다.. 참조값을 그대로 주면 원래 값도 수정되는 문제가 있었군요..